src/arizona_live_view.erl

```diff
@@ -26,38 +26,31 @@ Live view.
 %% API functions.
 -export([parse_str/2]).
 -ignore_xref([parse_str/2]).
--export([compile/3]).
--ignore_xref([compile/3]).
--export([persist_get/3]).
 -export([mount/2]).
 -export([handle_event/4]).
-
 -export([put_macro/3]).
 -ignore_xref([put_macro/3]).
 -export([get_macro/3]).
 -ignore_xref([get_macro/3]).
 
--opaque macros() :: map().
+-type macros() :: map().
 -export_type([macros/0]).
 
--opaque tree() :: list().
--export_type([tree/0]).
-
-%% Macros
--define(PERSIST_KEY, ?MODULE).
-
 %% --------------------------------------------------------------------
 %% Callbacks.
 %% --------------------------------------------------------------------
 
--callback mount(arizona_socket:t()) ->
-    {ok, arizona_socket:t()}.
+-callback mount(Socket) -> Socket
+    when Socket :: arizona_socket:t().
```

Comment:
Will this not consider that the input and output should be the same?
Shouldn't it be something like the below?
```suggestion
-callback mount(Socket0) -> Socket1
    when Socket0 :: arizona_socket:t(),
         Socket1 :: arizona_socket:t().
```

Reply:
No, it's just a spec. It considers them both to the of type :t

It's basically equivalent to 

```erlang
-callback mount(t()) -> t()
```

Mind you that in the context of specs there's no concept of variable. If that were the case we'd have much more stuff broken in this PR 😄

Reply:
For clarity I'll use this pattern everywhere (adding an `[n]` to a "changed" element). Mind you that the spec's don't care about that, though.

Reply:
[Erlang does say](https://www.erlang.org/docs/27/system/typespec.html#specifications-for-functions) "The latter specification says that the function takes some tuple and returns some tuple. The specification with the X type variable specifies that the function takes a tuple and returns **the same** tuple.

However, it is up to the tools that process the specifications to choose whether to take this extra information into account or not.", so I can only assume that the Dialyzer version we're using doesn't care for that extra information (?)

Reply:
No problem. If there is no issue, feel free to ignore my comments about this :slightly_smiling_face:

Reply:
a83a068